Link cookies page to terms of use and privacy policy

GemTay opened the pull request:
## Purpose
The ticket's original purpose was to add content to the cookie preferences pages, however this content already exists. Instead this ticket now add's links to the terms of use and privacy policy pages to take the user to the cookie preferences pages

UPDATE - this ticket now also adds a back button to the cookies page with the functionality to take the user back to the page they were on before they visited the cookies page

Fixes UML-896

## Checklist

* [x] I have performed a self-review of my own code
* [ ] I have added relevant logging with appropriate levels to my code
* [ ] I have updated documentation (Confluence/GitHub wiki/tech debt doc) where relevant
* [x] I have added tests to prove my work
* [ ] The product team have tested these changes

williamfalconeruk commented:
maybe add some behat tests to check that the links are all working

cooperaj commented:
You've done the test work now but it's worth clarifying that we shouldn't be testing content really. Also the step names are too specific in that we should be describing business actions not instructions. It's not "click the cookies link" but more like "I can navigate to cookie information"

GemTay commented:
I've refactored the referer validation steps and moved it into the get method (so that it is validated and ready to use on the back button). I added new function inside the `urlValidityCheckService` to return an appropriate url referer depending on the result of both functions `isValid` and `checkRefererRouteValid`. Since all this sanitisation is now done in the get method, no checks are done again in the post method. 

I'd like to get others opinions on whether the referer needs to be checked again in the post method? Since it's stored as a hidden value in the form, could it be manipulated by the user in anyway?

cooperaj commented:
> Since it's stored as a hidden value in the form, could it be manipulated by the user in anyway?

Never trust user input. A clever person would be able to craft a post submission that could conceivably give a back button that goes anywhere they want.

cooperaj commented on service-front/app/src/Common/src/Handler/CookiesPageHandler.php:
```diff
@@ -96,12 +88,7 @@ public function handlePost(ServerRequestInterface $request): ResponseInterface
-        if ($isValidRefererRoute) {
-            $response = new RedirectResponse($isValidRefererRoute ? $refererRoute : $this->urlHelper->generate('home'));
-        }
+        $response = new RedirectResponse($form->get('referer')->getValue());
```
Yup, you've removed all the safety here. Don't trust user input.

GemTay replied:
Pushing the referer through the validation service now

cooperaj commented on service-front/app/test/CommonTest/Service/Url/UrlValidityCheckServiceTest.php:
```diff
@@ -93,12 +102,59 @@ public function it_checks_if_referer_route_exists()
     }
+
+    /** @test */
+    public function it_returns_a_valid_referer()
```
Should probably also test for a non-valid referrer
